src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +77,1170 @@ ActorInit En_Slime_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FB60.s")
+    // Note: Does not match if you use this->actor instead of thisx
+    thisx->shape.shadowAlpha = this->iceBlockTimer;
+    if (this->iceBlockTimer == ICE_BLOCK_UNUSED) {
+        if (thisx->scale.y > 0.0001f) {
+            thisx->targetArrowOffset = 60.0f / thisx->scale.y;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FBA0.s")
+        if (this->collider.base.ocFlags1 & OC1_ON) {
+            Actor_MoveWithGravity(thisx);
+            Actor_UpdateBgCheckInfo(play, thisx, 20.0f, 35.0f, 40.0f, 0x1D);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FD94.s")
+        Collider_UpdateCylinder(thisx, &this->collider);
+        this->collider.dim.radius = sCylinderInit.dim.radius * (100.0f * thisx->scale.z);
+        this->collider.dim.height = sCylinderInit.dim.height * (100.0f * thisx->scale.y);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2FE38.s")
+        if (this->actionFunc == EnSlime_Jump) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30018.s")
+        if (this->collider.base.acFlags & AC_ON) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30344.s")
+        if (player->stateFlags3 & PLAYER_STATE3_100) {
+            this->collider.base.ocFlags1 &= ~OC1_TYPE_PLAYER;
+        } else {
+            this->collider.base.ocFlags1 |= OC1_TYPE_PLAYER;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30454.s")
+        if (this->collider.base.ocFlags1 & OC1_ON) {
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A304B8.s")
+        Actor_SetFocus(thisx, 15.0f);
+        if (this->drawDmgEffAlpha > 0.0f) {
+            if (this->drawDmgEffType != ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+                Math_StepToF(&this->drawDmgEffAlpha, 0.0f, 0.05f);
+                this->drawDmgEffScale = (this->drawDmgEffAlpha + 1.0f) * 0.2f;
+                this->drawDmgEffScale = CLAMP_MAX(this->drawDmgEffScale, 0.4f);
+            } else if (!Math_StepToF(&this->drawDmgEffFrozenSteamScale, 0.4f, 0.01f)) {
+                func_800B9010(thisx, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A3072C.s")
+void EnSlime_Draw(Actor* thisx, PlayState* play) {
+    s32 i;
+    EnSlime* this = THIS;
+    Vec3f wobbleScale;
+    Color_RGBA8* primColor;
+    Color_RGBA8* envColor;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A30778.s")
+    OPEN_DISPS(play->state.gfxCtx);
```

Comment:
Minor nit, but we like to have a blank line between the `OPEN_DISPS` and everything that follows.